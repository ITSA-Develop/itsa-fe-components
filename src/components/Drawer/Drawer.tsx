import { Drawer as AntDrawer, DrawerProps } from 'antd';

export const Drawer = ({ ...rest }: DrawerProps) => {
	return <AntDrawer {...rest} />;
};
