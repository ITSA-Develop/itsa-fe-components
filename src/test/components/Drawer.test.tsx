import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Drawer } from '../../components/Drawer/Drawer';

describe('Drawer component', () => {
	it('renders without crashing', () => {
		const { container } = render(<Drawer open={true} />);
		expect(document.querySelector('.ant-drawer')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('renders title and content', () => {
		const { container } = render(
			<Drawer open={true} title="Test Title">
				<div>Drawer content</div>
			</Drawer>,
		);
		expect(screen.getByText('Test Title')).toBeInTheDocument();
		expect(screen.getByText('Drawer content')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('calls onClose when close button is clicked', async () => {
		const onClose = vi.fn();
		render(<Drawer open={true} onClose={onClose} />);

		const closeBtn = document.querySelector('.ant-drawer-close');
		expect(closeBtn).toBeInTheDocument();

		if (closeBtn) {
			await userEvent.click(closeBtn);
		}

		expect(onClose).toHaveBeenCalled();
	});
});
