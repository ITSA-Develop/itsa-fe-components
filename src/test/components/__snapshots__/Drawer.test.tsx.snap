// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Drawer component > renders title and content 1`] = `<div />`;

exports[`Drawer component > renders without crashing 1`] = `<div />`;
